source/AI.cpp

```diff
@@ -3089,7 +3098,52 @@ double AI::RendezvousTime(const Point &p, const Point &v, double vp)
 	return numeric_limits<double>::quiet_NaN();
 }
 
+void AI::MoveWingman(Ship &ship, const PlayerInfo &player, Command &activeCommands)
```

Comment:
I would prefer a more generic MovePlayerShip(...) that can be used for the first player and all wingmen.

Reply:
I think this is doable but I have a question based on a bug I noticed in my implementation here: https://github.com/endless-sky/endless-sky/pull/5578/files#diff-8c4b03ae17df6f1471814c772c4f75e263c7e50b116a01bb3995c2f473961fd9R3135

I think this will cause problems because it is checking if the weapon to be used is the players selected weapon, which won't be the case for wingmen. Do you have any advice on where I can find the selected weapon for a generic ship/escort/etc?

Reply:
I didn't check, but I expect the AI ships to just fire any weapons they want to fire. A computer doesn't need to select weapons, such selection is only needed for players that have a limited set of input keys. We could argue why we only allow the player to fire 1 secondary weapon at a time, but that is for a different time and a different PR.

Giving different players their own weapon selection would probably require a split-up of the PlayerInfo class. It currently handles a lot of generic world/universe data and a lot of data that would be per-player.